keyboards/skdevices/adler/adler.c

```diff
@@ -0,0 +1,651 @@
+    {0, SW4_CS2,   SW5_CS2,   SW6_CS2}, //W
+    {0, SW4_CS3,   SW5_CS3,   SW6_CS3}, //E
+    {0, SW4_CS4,   SW5_CS4,   SW6_CS4}, //R
+    {0, SW4_CS5,   SW5_CS5,   SW6_CS5}, //T
+    {0, SW4_CS6,   SW5_CS6,   SW6_CS6}, //Y
+    {0, SW4_CS7,   SW5_CS7,   SW6_CS7}, //U
+    {0, SW4_CS8,   SW5_CS8,   SW6_CS8}, //I
+    {0, SW4_CS9,   SW5_CS9,   SW6_CS9}, //O
+    {0, SW4_CS10,   SW5_CS10,   SW6_CS10}, //P
+    {0, SW4_CS11,   SW5_CS11,   SW6_CS11}, //[
+    {0, SW4_CS12,   SW5_CS12,   SW6_CS12}, //]
+    {0, SW4_CS13,   SW5_CS13,   SW6_CS13}, //|
+    {0, SW4_CS14,   SW5_CS14,   SW6_CS14}, //Ins
+    {0, SW4_CS15,   SW5_CS15,   SW6_CS15}, //Home
+    {0, SW4_CS16,   SW5_CS16,   SW6_CS16}, //PgUp
+    
+    {1, SW4_CS3,   SW5_CS3,   SW6_CS3}, //M3
+    {1, SW1_CS4,   SW2_CS4,   SW3_CS4}, //Caps
+    {0, SW1_CS1,   SW2_CS1,   SW3_CS1}, //A
+    {0, SW1_CS2,   SW2_CS2,   SW3_CS2}, //S
+    {0, SW1_CS3,   SW2_CS3,   SW3_CS3}, //D
+    {0, SW1_CS4,   SW2_CS4,   SW3_CS4}, //F
+    {0, SW1_CS5,   SW2_CS5,   SW3_CS5}, //G
+    {0, SW1_CS6,   SW2_CS6,   SW3_CS6}, //H
+    {0, SW1_CS7,   SW2_CS7,   SW3_CS7}, //J
+    {0, SW1_CS8,   SW2_CS8,   SW3_CS8}, //K
+    {0, SW1_CS9,   SW2_CS9,   SW3_CS9}, //L
+    {0, SW1_CS10,   SW2_CS10,   SW3_CS10}, //;
+    {0, SW1_CS11,   SW2_CS11,   SW3_CS11}, //'
+    {0, SW1_CS13,   SW2_CS13,   SW3_CS13}, //Enter
+    {0, SW1_CS14,   SW2_CS14,   SW3_CS14}, //Del
+    {0, SW1_CS15,   SW2_CS15,   SW3_CS15}, //End
+    {0, SW1_CS16,   SW2_CS16,   SW3_CS16}, //PgDown
+
+    {1, SW4_CS4,   SW5_CS4,   SW6_CS4}, //M4
+    {1, SW1_CS5,   SW2_CS5,   SW3_CS5}, //LShift
+    {1, SW10_CS2,   SW11_CS2,   SW12_CS2}, //Z
+    {1, SW10_CS3,   SW11_CS3,   SW12_CS3}, //X
+    {1, SW10_CS4,   SW11_CS4,   SW12_CS4}, //C
+    {1, SW10_CS5,   SW11_CS5,   SW12_CS5}, //V
+    {1, SW10_CS6,   SW11_CS6,   SW12_CS6}, //B
+    {1, SW10_CS7,   SW11_CS7,   SW12_CS7}, //N
+    {1, SW10_CS8,   SW11_CS8,   SW12_CS8}, //M
+    {1, SW10_CS9,   SW11_CS9,   SW12_CS9}, //,
+    {1, SW10_CS10,   SW11_CS10,   SW12_CS10}, //.
+    {1, SW10_CS11,   SW11_CS11,   SW12_CS11}, ///
+    {1, SW10_CS13,   SW11_CS13,   SW12_CS13}, //RShift
+    {1, SW10_CS15,   SW11_CS15,   SW12_CS15}, //Up
+
+    {1, SW4_CS5,   SW5_CS5,   SW6_CS5}, //M5
+    {1, SW1_CS6,   SW2_CS6,   SW3_CS6}, //LCtrl
+    {1, SW7_CS1,   SW8_CS1,   SW9_CS1}, //Win
+    {1, SW7_CS2,   SW8_CS2,   SW9_CS2}, //LAlt
+    {1, SW7_CS6,   SW8_CS6,   SW9_CS6}, //Space
+    {1, SW7_CS10,   SW8_CS10,   SW9_CS10}, //RAlt
+    {1, SW7_CS11,   SW8_CS11,   SW9_CS11}, //Fn
+    {1, SW7_CS12,   SW8_CS12,   SW9_CS12}, //Menu
+    {1, SW7_CS13,   SW8_CS13,   SW9_CS13}, //RCtrl
+    {1, SW7_CS14,   SW8_CS14,   SW9_CS14}, //Left
+    {1, SW7_CS15,   SW8_CS15,   SW9_CS15}, //Down
+    {1, SW7_CS16,   SW8_CS16,   SW9_CS16}, //Right
+};
+
+#ifdef OLED_ENABLE
+static bool animation_finished = false;
+#define FRAME_DURATION 333
+uint32_t timer = 0;
+uint8_t current_frame = 0;
+
+static void render_logo(void) {
```

Comment:
You could reduce the size significantly by removing the columns that are empty.   I know it's not as much of a concern here (f411), but may be a good idea, in general.